life_expectancy/cleaning.py

```diff
@@ -1,47 +1,71 @@
 """
 Script to load, clean and save data
 """
-import pandas as pd
 import argparse
 import pathlib
+import pandas as pd
 
-# life_path = pathlib.Path.cwd()
-# data_path = life_path.joinpath('data')
 
-def clean_data(path: str, country: str = 'PT'):
+life_path = pathlib.Path(__file__).parent
+data_path = life_path / 'data'
+FILE_NAME = 'eu_life_expectancy_raw.tsv'
+
+def load_data(file_name: str)-> pd.DataFrame:
     """
-    Function to clean data
+        Function to load data from data folder, given a file_name
     """
+    df_expectancy = pd.read_csv(data_path / file_name, sep = '\t')
+    return df_expectancy
 
-    df_expectancy = pd.read_csv(path, sep = '\t') # data_path.joinpath(file_name)
-
+def clean_data(df_expectancy: pd.DataFrame, country: str = 'PT') -> pd.DataFrame:
+    """
+        Function to clean data
+    """
+    # Splitting the name of the first column by ','
     new_cols = df_expectancy.columns[0].split(',')
 
+    # Splitting the data of the first column and dropping the column with a bad format
     df_expectancy[new_cols] = df_expectancy[df_expectancy.columns[0]].str.split(',', expand=True)
 
     df_expectancy.drop(df_expectancy.columns[0], axis = 1, inplace = True)
 
+    # Creating a table with unit,sex,age,region,year,value as columns
     df_simple = df_expectancy.melt(id_vars = new_cols, var_name = 'year', value_name = 'value')
 
     df_simple.rename(columns={'geo\\time': 'region'}, inplace=True)
 
     df_simple['year'].astype('int')
 
+    # allowing only floats as values to 'value' column
     df_simple['value'] = (df_simple['value'].str.extract(r'(\d+\.?\d*)').astype(float))
     df_simple = df_simple.dropna(subset=['value'])
 
     df_simple.dropna(subset=['value'], inplace = True)
 
-    df_pt = df_simple[df_simple['region'] == country]
+    # filtering by the desired country
+    df_country = df_simple[df_simple['region'] == country]
 
-    df_pt.to_csv('life_expectancy/data/pt_life_expectancy.csv', index = False)
+    return df_country
+
+def save_data(df_country: pd.DataFrame, file_name: str, file_path: pathlib.Path) -> None:
+    """
+        Function to save data, given a file_name and a path
+    """
+    df_country.to_csv(file_path / file_name, index = False)
+
+
+def main(country: str) -> None:
+    """
+        Function that loads, cleans and saves data
+    """
+    data_df = load_data(FILE_NAME)
+    clean_df = clean_data(df_expectancy = data_df, country = country)
+    save_data(df_country = clean_df, file_name = 'pt_life_expectancy.csv', file_path = data_path)
```

Comment:
Suggestion: I consider this to be nitpicking, but to maintain code cohesion and logic I would add an INPUT_FILE_NAME and an OUTPUT_FILE_NAME in the global variables.

----- Example -----
INPUT_FILE_NAME = 'eu_life_expectancy_raw.tsv'
OUTPUT_FILE_NAME = 'pt_life_expectancy.csv'
data_df = load_data(INPUT_FILE_NAME)
save_data(df_country = clean_df, file_name = OUPUT_FILE_NAME, file_path = data_path)